packages/p2p-media-loader-hlsjs/src/engine.ts

```diff
@@ -10,40 +10,53 @@ import type { HlsConfig, Events } from "hls.js";
 import { FragmentLoaderBase } from "./fragment-loader";
 import { PlaylistLoaderBase } from "./playlist-loader";
 import { SegmentManager } from "./segment-mananger";
-import { Core, CoreEventMap } from "p2p-media-loader-core";
+import {
+  CoreConfig,
+  Core,
+  CoreEventMap,
+  DynamicConfig,
+} from "p2p-media-loader-core";
 
 export class Engine {
   private readonly core: Core;
   private readonly segmentManager: SegmentManager;
   private hlsInstanceGetter?: () => Hls;
   private currentHlsInstance?: Hls;
 
-  constructor() {
-    this.core = new Core();
+  constructor(private readonly config?: CoreConfig) {
+    this.core = new Core(config);
     this.segmentManager = new SegmentManager(this.core);
   }
 
-  public addEventListener<K extends keyof CoreEventMap>(
+  addEventListener<K extends keyof CoreEventMap>(
     eventName: K,
     listener: CoreEventMap[K],
   ) {
     this.core.addEventListener(eventName, listener);
   }
 
-  public removeEventListener<K extends keyof CoreEventMap>(
+  removeEventListener<K extends keyof CoreEventMap>(
     eventName: K,
     listener: CoreEventMap[K],
   ) {
     this.core.removeEventListener(eventName, listener);
   }
 
-  public getConfig(): Partial<HlsConfig> {
+  getHlsConfig(): Partial<HlsConfig> {
     return {
       fLoader: this.createFragmentLoaderClass(),
       pLoader: this.createPlaylistLoaderClass(),
     };
   }
 
+  getConfig() {
+    return this.config;
```

Comment:
removed `this.core.config`
But after applying dynamic config we won't get changed config. Is this OK?